### PR TITLE
Add SCC for clusterController on OpenShift

### PR DESCRIPTION
**Pull request by ivankube**
## What does this PR change?
Currently OpenShift users can't add clusterController because of

`W1108 14:19:26.493318   42854 warnings.go:70] would violate PodSecurity "restricted:v1.24": hostPort (container "kubecost-cluster-controller" uses hostPort 9731), allowPrivilegeEscalation != false (container "kubecost-cluster-controller" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "kubecost-cluster-controller" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "kubecost-cluster-controller" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "kubecost-cluster-controller" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")`

## Does this PR rely on any other PRs?
No

## How does this PR impact users? (This is the kind of thing that goes in release notes!)
Add SecurityContext for clusterController when it's enabled on OpenShift

## Links to Issues or tickets this PR addresses or fixes
https://kubecost.atlassian.net/browse/SUP-6255

## What risks are associated with merging this PR? What is required to fully test this PR?
Minimal risks.

## How was this PR tested?
Manually

## Have you made an update to documentation? If so, please provide the corresponding PR.
N/A

### CODE REVIEW

File: cost-analyzer/templates/cloud-controller-ocp-scc.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+seLinuxContext:
+  type: RunAsAny
+supplementalGroups:
+  type: RunAsAny
+seccompProfiles:
+- runtime/default
+volumes:
+  - hostPath
+  - projected
+  - configMap
```
Comment (thomasvn) [inline cost-analyzer/templates/cloud-controller-ocp-scc.yaml:27]:
```suggestion
volumes:
  - hostPath
  - projected
  - configMap
  - secret
```

According to [SUP-6255](https://kubecost.atlassian.net/browse/SUP-6255) `secret` is required here.

**Comment by thomasvn** [comment]
@ivankube Generally this looks good! Could you please resolve the merge conflicts here? And can you also provide any documentation you have on how this was tested?

**Comment by ivankube** [comment]
@thomasvn thank you for the review!

I tested it manually by creating a new local helm chart and using this chart with the Replicated Openshift cluster.

**Review by chipzoller — CHANGES_REQUESTED** [review]
Please also include an update in the CI workflow [here](https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/.github/workflows/chart.yaml#L184-L190) to enable this new SCC and Cluster Controller.

**Comment by ivankube** [comment]
@chipzoller https://github.com/kubecost/cost-analyzer-helm-chart/pull/3785

**Comment by chipzoller** [comment]
Why didn't you just include those changes as part of this PR?

**Comment by ivankube** [comment]
@chipzoller I've just added the changes to this PR and closed 3785.

**Review by chipzoller — APPROVED** [review]

**Pull request merged**